src/ape_geth/provider.py

```diff
@@ -197,9 +199,9 @@ def wait(self, *args, **kwargs):
 
 class GethNetworkConfig(PluginConfig):
     # Make sure you are running the right networks when you try for these
-    mainnet: Dict = DEFAULT_SETTINGS.copy()
-    goerli: Dict = DEFAULT_SETTINGS.copy()
-    sepolia: Dict = DEFAULT_SETTINGS.copy()
+    mainnet: Dict = {"uri": random.choice(PUBLIC_CHAIN_RPCS["ethereum"]["mainnet"])}
+    goerli: Dict = {"uri": random.choice(PUBLIC_CHAIN_RPCS["ethereum"]["goerli"])}
+    sepolia: Dict = {"uri": random.choice(PUBLIC_CHAIN_RPCS["ethereum"]["sepolia"])}
     # Make sure to run via `geth --dev` (or similar)
     local: Dict = {**DEFAULT_SETTINGS.copy(), "chain_id": DEFAULT_TEST_CHAIN_ID}
 
```

Comment:
we have to handle the non-eth networks via a custom validator or something

Reply:
Couple of questions.  Are we responsible for non-eth networks here?  Or would the ecosystem plugin handle that?  I don't think your change allows for custom ecosystems.  Though maybe I should test against one?

I wrote a little validator for the `GethNetworkConfig` but since it's only used for Ethereum, it's basically pointless so I'm not planning on committing it.

```python
    @model_validator(mode="after")
    def verify_networks(self) -> "GethEcosystemConfig":
        if self.__pydantic_extra__:
            for aname in self.__pydantic_extra__.keys():
                rpcs = PUBLIC_CHAIN_RPCS["ethereum"].get(aname)
                if rpcs:
                    self.__pydantic_extra__[aname] = {"uri": random.choice(rpcs)}
                    continue
                # Fallback to default settings.
                self.__pydantic_extra__[aname] = {**DEFAULT_SETTINGS.copy()}
        return self
```

So, is any change necessary here?

Reply:
Or maybe you're not talking about custom networks.  I'll test against ape_optimism or something.

Reply:
i think both custom networks and the plugins use `GethNetworkConfig`

Reply:
Yeah, I don't think custom networks are relevant though since they won't ever match a public RPC.  Working on optimism now though.  Should have something shortly.

Reply:
I updated the `Geth` provider to handle dynamic networks.  Tiny bit of code duplication with `EthereumNodeProvider` but I didn't want to fully override it either.

Reply:
I think that works